src/btree/bt_split.c

```diff
@@ -1030,6 +1030,16 @@ __split_parent(WT_SESSION_IMPL *session, WT_REF *ref,
 #endif
 
 	/*
+	 * Page-delete information is only read when the WT_REF state is
+	 * WT_REF_DELETED.  The page-delete memory wasn't added to the
+	 * parent's footprint, ignore it here.
+	 */
+	if (ref->page_del != NULL) {
```

Comment:
Thanks, @agorrod.  I think this is correct.  We are replacing or discarding `ref`.  Either way, we should free `page_del` if there is one still attached.